src/scancode_config.py

```diff
@@ -0,0 +1,156 @@
+#
+# Copyright (c) 2018 nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/scancode-toolkit/
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import errno
+import os
+from os.path import abspath
+from os.path import dirname
+from os.path import expanduser
+from os.path import join
+from os.path import exists
+import tempfile
+
+"""
+Core configuration globals.
+
+Note: this module MUST import ONLY from the standard library.
+"""
+
+# this exception is not available on posix
+try:
+    WindowsError  # noqa
+except NameError:
+    WindowsError = None  # NOQA
+
+
+def _create_dir(location):
+    """
+    Create directory and all sub-directories recursively at `location`.
+    Raise Exceptions if it fails to create the directory.
+    NOTE: this is essentailly a copy of commoncode.fileutils.create_dir()
```

Comment:
The function is slightly different from `commoncode.fileutils.create_dir()`. In particular, it does not call `fsencode()` for dealing with unicode issues on Linux. Is this intentional? If not, how about we get rid of the duplication by importing `_create_dir()` in `commoncode.fileutils`?

Reply:
https://guides.github.com/features/mastering-markdown/You have eagle eyes! I did not want that module to depend on anything but the stdlib. Therefore I copied and duplicated the code from commoncode AND removed the dep on the fsencode from os.backports.... But now you are asking, why did I do so? there is probably no seriously good reason, except to make sure there would be no circular imports? Actually this is the reason: https://github.com/nexB/scancode-toolkit/blob/5784b20a5c4263a55b8efae0509767aca30caeea/src/commoncode/fileutils.py#L55 ... I guess I could add some comment about this weirdness.

Reply:
because fileutils also imports scancode_config